File: GGAndroid_2.0/GGPay/pay-alipay/src/main/java/io/ganguo/pay/alipay/PayResultHandler.java
package io.ganguo.pay.alipay;

import android.os.Handler;
import android.os.Message;
import android.util.Log;

import java.lang.ref.WeakReference;

import io.ganguo.pay.core.GGPay;
import io.ganguo.pay.core.IPayService;
import io.ganguo.pay.core.PayCallback;
import io.ganguo.pay.core.PayResult;

/**
 * Created by Roger on 05/07/17.
 */
final class PayResultHandler extends Handler {
    private PayCallback mCallback;
    private WeakReference<IPayService> ref;

    PayResultHandler(PayCallback payCallback, IPayService IPayService) {
        this.mCallback = payCallback;
        this.ref = new WeakReference<IPayService>(IPayService);
    }

    /**
     * 处理支付结果回调
     */
    @Override
    public void handleMessage(Message msg) {
        super.handleMessage(msg);
        switch (msg.what) {
            case AlipayService.PAY_FLAG:
                PayResult payResult = new PayResult();
                @SuppressWarnings("unchecked")
                PayData aliPayData = new PayData(msg.obj.toString());
                String resultStatus = aliPayData.getResultStatus();
                payResult.type = "alipay";
                payResult.result = aliPayData.getResult();
                payResult.code = resultStatus;
                payResult.message = aliPayData.getMemo();
                Log.e("TAG", "handleMessage:payStatus: " + resultStatus);
                Log.e("TAG", "handleMessage: " + msg.obj.toString());
                if (mCallback == null) {
                    final IPayService service = ref.get();
                    if (service != null) {
                        GGPay.remove(service);
                    }
                    return;
                }
                if (isEquals(resultStatus, AlipayService.PAY_OK)) {
                    mCallback.onPaySuccess(payResult);
                } else if (isEquals(resultStatus, AlipayService.PAY_CANCEL)) {
                    mCallback.onPayCancel(payResult);
                } else {
                    mCallback.onPayError(payResult);
                }
                mCallback = null;
                final IPayService service = ref.get();
                if (service != null) {
                    GGPay.remove(service);
                }
        }
    }

    private boolean isEquals(String s, String target) {
        return s != null && target != null && s.equals(target);
    }

    void release() {
        mCallback = null;
    }
}

File: GGAndroid_4.0/core/service/service-pay/pay-alipay/src/main/kotlin/io/ganguo/pay/alipay/PayResultHandler.kt
package io.ganguo.pay.alipay

import android.os.Handler
import android.os.Message
import android.util.Log
import io.ganguo.factory.GGFactory
import io.ganguo.pay.core.annotation.PayStatus
import io.ganguo.pay.core.annotation.PayType

/**
 * Created by Roger on 05/07/17.
 * 支付结果回调处理
 */
internal class PayResultHandler : Handler() {

    /**
     * 处理支付结果回调
     * @param msg
     */
    override fun handleMessage(msg: Message) {
        super.handleMessage(msg)
        if (msg.what != AliPayConstants.PayStatusCode.PAY_FLAG) {
            return
        }
        val payResult = AliPayResult(PayType.ALI_PAY)
        val aliPayData = AliPayOrderInfo(msg.obj.toString())
        val resultStatus = aliPayData.resultStatus

        payResult.result = aliPayData
        payResult.code = resultStatus
        payResult.message = aliPayData.memo
        payResult.status = transformStatus(payResult.code)

        GGFactory.asRxResultSend<AliPayResult>(AliPayMethod::class.java)
                ?.sendResult(payResult)

        Log.d(javaClass.simpleName, "handleMessage:payStatus: $resultStatus")
        Log.d(javaClass.simpleName, "handleMessage: " + msg.obj.toString())
    }


    /**
     * 判断支付状态
     * @param resultCode Int
     * @return PayStatus
     */
    @PayStatus
    private fun transformStatus(resultCode: Int): Int {
        return when (resultCode) {
            AliPayConstants.PayStatusCode.PAY_OK -> PayStatus.SUCCESS
            AliPayConstants.PayStatusCode.PAY_CANCEL -> PayStatus.CANCEL
            else -> PayStatus.FAILED
        }
    }

}
